org.sonarlint.eclipse.core/src/org/sonarlint/eclipse/core/internal/jobs/AnalyzeProjectJob.java

```diff
@@ -310,86 +351,64 @@ private static RawIssueTrackable transform(Issue issue, IResource resource, IDoc
-    if (serverModuleKey == null) {
-      // not bound to a module -> nothing to do
-      return tracked;
-    }
-
-    Server server = (Server) ServersManager.getInstance().getServer(serverId);
+  private Collection<Trackable> trackServerIssuesSync(Server server, ISonarLintFile resource, Collection<Trackable> tracked) {
     ServerConfiguration serverConfiguration = server.getConfig();
     ConnectedSonarLintEngine engine = server.getEngine();
-    List<ServerIssue> serverIssues = ServerIssueUpdater.fetchServerIssues(serverConfiguration, engine, serverModuleKey, resource);
+    List<ServerIssue> serverIssues = ServerIssueUpdater.fetchServerIssues(serverConfiguration, engine, getProjectConfig().getModuleKey(), resource);
     Collection<Trackable> serverIssuesTrackable = serverIssues.stream().map(ServerIssueTrackable::new).collect(Collectors.toList());
     return IssueTracker.matchAndTrackServerIssues(serverIssuesTrackable, tracked);
   }
 
-  private void trackServerIssuesAsync(Collection<IResource> resources) {
-    String serverId = getSonarProject().getServerId();
-
-    if (serverId == null) {
-      // not bound to a server -> nothing to do
-      return;
-    }
-
-    String serverModuleKey = SonarLintCorePlugin.getDefault().getProjectManager().readSonarLintConfiguration(this.getSonarProject().getProject()).getModuleKey();
-    if (serverModuleKey == null) {
-      // not bound to a module -> nothing to do
-      return;
-    }
-
-    Server server = (Server) ServersManager.getInstance().getServer(serverId);
+  private void trackServerIssuesAsync(Server server, Collection<ISonarLintIssuable> resources) {
     ServerConfiguration serverConfiguration = server.getConfig();
     ConnectedSonarLintEngine engine = server.getEngine();
-    String localModuleKey = getSonarProject().getProject().getName();
-    SonarLintCorePlugin.getDefault().getServerIssueUpdater().updateAsync(serverConfiguration, engine, getSonarProject(), localModuleKey, serverModuleKey, resources);
+    String localModuleKey = getProject().getName();
+    SonarLintCorePlugin.getDefault().getServerIssueUpdater().updateAsync(serverConfiguration, engine, getProject(), localModuleKey, getProjectConfig().getModuleKey(), resources);
   }
 
-  private static void analysisCompleted(Collection<ProjectConfigurator> usedConfigurators, Map<String, String> properties, final IProgressMonitor monitor) {
-    for (ProjectConfigurator p : usedConfigurators) {
-      p.analysisComplete(Collections.unmodifiableMap(properties), monitor);
+  private static void analysisCompleted(Collection<ProjectConfigurator> usedDeprecatedConfigurators, Collection<IAnalysisConfigurator> usedConfigurators,
+    Map<String, String> properties, final IProgressMonitor monitor) {
+    Map<String, String> unmodifiableMap = Collections.unmodifiableMap(properties);
+    for (ProjectConfigurator p : usedDeprecatedConfigurators) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Remove this use of "ProjectConfigurator"; it is deprecated. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3ACallToDeprecatedMethod)